model/flow/protocol_state.go

```diff
@@ -59,6 +60,20 @@ type EpochStateContainer struct {
 	// epoch are only allowed to listen to the network but not actively contribute. Such
 	// nodes are _not_ part of `Identities`.
 	ActiveIdentities DynamicIdentityEntryList
+
+	// EpochExtensions represents extensions to the last successful epoch in EFM. In the happy path it is nil.
+	// Epochs in which EFM is triggered will have at least one EpochExtension.
+	// Extensions is an ordered list, such that for each consecutive pair of
+	// EpochExtensions e1, e2, e1.FinalView+1 = e2.FirstView.
+	EpochExtensions []EpochExtension
+}
+
+// EpochExtension represents a range of views, which contiguously extends the
+// current epoch E.
```

Comment:
I think we have a notion of "current Epoch": the epoch that the view with this Protocol State belongs to. In comparison, the `EpochExtension` belongs to the epoch described by the `EpochStateContainer`, which might be the previous or the current epoch. 

```suggestion
// EpochExtension represents a range of views, which contiguously extends this epoch.
```